src/ansys/aedt/core/workflows/hfss3dlayout/via_clustering_extension.py

```diff
@@ -0,0 +1,262 @@
+        master.flag = True
+        hfss = Hfss3dLayout(
+            new_desktop=False,
+            version=version,
+            port=port,
+            aedt_process_id=aedt_process_id,
+            student_version=is_student,
+        )
+        hfss.save_project()
+        primitives = hfss.modeler.objects_by_layer(layer="via_merging")
+        if not primitives:
+            messagebox.showwarning(message="No primitives found on layer defined for merging padstack instances.")
+            hfss.release_desktop(close_desktop=False, close_projects=False)
+        else:
+            frontend_dict["contour_list"] = []
+            for primitive in primitives:
+                prim = hfss.modeler.geometries[primitive]
+                if prim.prim_type == "poly" or prim.prim_type == "rect":
+                    pts = [pt for pt in [_pt.position for _pt in prim.points]]
+                    frontend_dict["contour_list"].append(pts)
+                else:
+                    hfss.logger.warning(
+                        f"Unsupported primitive {prim.name}, only polygon and rectangles are supported."
+                    )
+        frontend_dict["new_aedb_path"] = os.path.join(active_project_path, project_name.get() + ".aedb")
+        master.destroy()
+
+    button_add_layer = ttk.Button(master, text="Add layer", width=40, command=add_drawing_layer, style="PyAEDT.TButton")
+    button_add_layer.grid(row=3, column=0, pady=10, padx=10)
+
+    button_merge_pdstacks = ttk.Button(
+        master, text="Merge padstack instances", width=40, command=callback, style="PyAEDT.TButton"
+    )
+    button_merge_pdstacks.grid(row=4, column=0, pady=10, padx=10)
+
+    button_frame = ttk.Frame(master, style="PyAEDT.TFrame", relief=tkinter.SUNKEN, borderwidth=2)
+    button_frame.grid(row=5, column=3, pady=10, padx=10)
+
+    # Add the toggle theme button inside the frame
+    change_theme_button = ttk.Button(
+        button_frame, width=10, text="\u263D", command=toggle_theme, style="PyAEDT.TButton"
+    )
+
+    change_theme_button.grid(row=0, column=0, padx=0)
+
+    tkinter.mainloop()
+    return frontend_dict
+
+
+def main(extension_arguments):
+    start_layer = extension_arguments.get("start_layer", False)
+    stop_layer = extension_arguments.get("stop_layer", False)
+    design_name = extension_arguments.get("design_name", "")
+    # nets_filter = extension_arguments.get("nets_filter", "")
+    contour_list = extension_arguments.get("contour_list", [])
+    aedb_path = extension_arguments.get("aedb_path", "")
+    new_aedb_path = extension_arguments.get("new_aedb_path", "")
+    shutil.copytree(aedb_path, new_aedb_path)
+    edb = Edb(new_aedb_path, design_name, edbversion=version)
+    edb.padstacks.merge_via(contour_boxes=contour_list, net_filter=None, start_layer=start_layer, stop_layer=stop_layer)
+    [prim.delete() for prim in edb.modeler.primitives_by_layer["via_merging"]]
```

Comment:
In this case, Codacy prefers:
```suggestion
    for prim in edb.modeler.primitives_by_layer["via_merging"]:
    prim.delete()
```